pkg/apply/merge_test.go

```diff
@@ -326,3 +326,115 @@ metadata:
 		})
 	})
 })
+
+var _ = Describe("MergeWebhookConfiguration", func() {
+	Context("when current has non empty casBundle and update is empty", func() {
+		current := k8s.UnstructuredFromYaml(`
+apiVersion: admissionregistration.k8s.io/v1beta1
+kind: MutatingWebhookConfiguration
+metadata:
+  name: nmstate
+  labels:
+    app: kubernetes-nmstate
+webhooks:
+  - name: nodenetworkconfigurationpolicies-mutate.nmstate.io
+    clientConfig:
+      caBundle: nodenetworkconfigurationpolicies-mutate.nmstate.io-ca
+      service:
+        name: nmstate-webhook
+        namespace: nmstate
+        path: "/nodenetworkconfigurationpolicies-mutate"
+    rules:
+      - operations: ["CREATE", "UPDATE"]
+        apiGroups: ["*"]
+        apiVersions: ["v1alpha1"]
+        resources: ["nodenetworkconfigurationpolicies"]
+  - name: nodenetworkconfigurationpolicies-status-mutate.nmstate.io
+    clientConfig:
+      caBundle: nodenetworkconfigurationpolicies-status-mutate.nmstate.io-ca
+      service:
+        name: nmstate-webhook
+        namespace: nmstate
+        path: "/nodenetworkconfigurationpolicies-status-mutate"
+    rules:
+      - operations: ["CREATE", "UPDATE"]
+        apiGroups: ["*"]
+        apiVersions: ["v1alpha1"]
+        resources: ["nodenetworkconfigurationpolicies/status"]
+  - name: nodenetworkconfigurationpolicies-timestamp-mutate.nmstate.io
+    clientConfig:
+      caBundle: nodenetworkconfigurationpolicies-timestamp-mutate.nmstate.io-ca
+      service:
+        name: nmstate-webhook
+        namespace: nmstate
+        path: "/nodenetworkconfigurationpolicies-timestamp-mutate"
+    rules:
+      - operations: ["CREATE", "UPDATE"]
+        apiGroups: ["*"]
+        apiVersions: ["v1alpha1"]
+        resources: ["nodenetworkconfigurationpolicies", "nodenetworkconfigurationpolicies/status"]
+`)
+		updated := k8s.UnstructuredFromYaml(`
```

Comment:
It is impossible to spot the difference. And the test itself is little hard to decode for me as well.

Would you please keep a single MutatingWebhookConfiguration as a template and then populate it in before each? The constants you use to populate it may be then used in assertions, so it is clear what needs to change and what remain the same.